### PR TITLE
HOTFIX: use Jacksons string encoder for JSON output

### PR DESCRIPTION
**Pull request by i386**
# Description
We need to get something in here until we've got a real fix upstream in Stapler.

@kohsuke we should discuss the role of Stapler in JSON generation. It doesn't quite work and I think we would be better off using Jackson.

# Steps to reproduce

1. Run `https://github.com/i386/colortest`
2. Steps don't load due to invalid characters in the step name

# What you should see

After running this PR

screencapture-localhost-8080-jenkins-blue-organizations-jenkins-i386-2fnyancat-detail-master-3-pipeline-1503310468752

See [JENKINS-45099](https://issues.jenkins-ci.org/browse/JENKINS-45099).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by vivek** [comment]
@i386 I have added a test for json encoded name.

File: blueocean-commons/src/main/java/io/jenkins/blueocean/commons/stapler/export/JSONDataWriter.java [hunk 1]
```diff
@@ -103,17 +105,7 @@ public void valuePrimitive(Object v) throws IOException {
...
-            switch(c) {
-                case '"':   buf.append("\\\"");break;
-                case '\\':  buf.append("\\\\");break;
-                case '\n':  buf.append("\\n");break;
-                case '\r':  buf.append("\\r");break;
-                case '\t':  buf.append("\\t");break;
-                default:    buf.append(c);break;
-            }
-        }
+        JsonStringEncoder.getInstance().quoteAsString(v, buf);
```
Comment (i386) [inline blueocean-commons/src/main/java/io/jenkins/blueocean/commons/stapler/export/JSONDataWriter.java]: Sounds good to me

Comment (i386) [inline blueocean-commons/src/main/java/io/jenkins/blueocean/commons/stapler/export/JSONDataWriter.java]: Updated

**Comment by i386** [comment]
@vivek ready.

**Comment by vivek** [comment]
@i386 LGTM 🐝  waiting for CI to finish then will be good to merge.

**Pull request merged**